acestep/ui/gradio/events/generation/external_lm_setup_actions.py

```
@@ -0,0 +1,190 @@
+        profile = get_profile(provider)
+        resolved_base_url, models = discover_models(
+            profile=profile,
+            base_url=base_url,
+            api_key=api_key,
+        )
+        save_cache(provider=profile.provider_id, protocol=profile.protocol, base_url=resolved_base_url, models=models)
+    except ValueError as exc:
+        return gr.update(), status_formatter([f"❌ {exc}"])
+    except ExternalModelDiscoveryError as exc:
+        return gr.update(), status_formatter([t("messages.external_lm_endpoint_failed", error=str(exc))])
+    except Exception as exc:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)